pkg/gatherers/clusterconfig/gather_lokistack.go

```diff
@@ -0,0 +1,137 @@
+package clusterconfig
+
+// nolint: dupl
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+const LokiStackResourceLimit = 20
+
+// GatherOpenshiftLogging Collects `clusterlogging.logging.openshift.io` resources.
+//
+// ### API Reference
+// - https://github.com/openshift/cluster-logging-operator/blob/master/pkg/apis/logging/v1/clusterlogging_types.go
+//
+// ### Sample data
+// - docs/insights-archive-sample/config/logging/openshift-logging/instance.json
+//
+// ### Location in archive
+// - `namespace/openshift-logging/group/resource/{name}.json`
+//
+// ### Config ID
+// `clusterconfig/openshift_logging`
+//
+// ### Released version
+// - 4.18.0
+//
+// ### Backported versions
+// None
+//
+// ### Changes
+// None
+func (g *Gatherer) GatherLokiStack(ctx context.Context) ([]record.Record, []error) {
+	gatherDynamicClient, err := dynamic.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return gatherLokiStack(ctx, gatherDynamicClient)
+}
+
+func gatherLokiStack(ctx context.Context, dynamicClient dynamic.Interface) ([]record.Record, []error) {
+	klog.V(2).Info("Start LokiStack gathering")
+	loggingResourceList, err := dynamicClient.Resource(lokiStackResource).
+		Namespace("openshift-logging").
+		List(ctx, metav1.ListOptions{Limit: LokiStackResourceLimit})
+
+	if errors.IsNotFound(err) {
+		return nil, nil
+	}
+	if err != nil {
+		klog.V(2).Infof("Unable to list %s resource due to: %s", lokiStackResource, err)
+		return nil, []error{err}
+	}
+
+	var records []record.Record
+	var errs []error
+
+	numItems := len(loggingResourceList.Items)
+	if numItems > 1 {
+		errs = append(errs, fmt.Errorf("Found more resources than expected (expected 1)"))
```

Comment:
I think we don't need an error if there's more than one resource. On the other we need an error if there are some resources in different namespaces than `openshift-logging`

Reply:
I refactored the method to add an error when a `LokiStack` in a different namespace is found